TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -422,11 +422,34 @@ public override bool Visit(DataDescription dataDescription)
         {
             DataDescriptionEntry dataDescriptionEntry = dataDescription.CodeElement;
 
-            var levelNumber = dataDescriptionEntry.LevelNumber;
             //Check if the DataDescription is an empty group
-            if (levelNumber != null && dataDescription.IsDataDescriptionGroup && dataDescription.ChildrenCount == 0)
+            if (dataDescriptionEntry.LevelNumber != null && dataDescription.IsDataDescriptionGroup && dataDescription.ChildrenCount == 0)
             {
-                DiagnosticUtils.AddError(dataDescription, "A group item cannot be empty.", dataDescriptionEntry);
+                //Get all sibling nodes
+                var siblingNodes = dataDescription.Parent.Children.ToList();
+                //Get current node index
+                var nodeIndex = siblingNodes.IndexOf(dataDescription);
```

Comment:
No need to create a list here, the method `IndexOf` is already available on `Node` class so you can retrieve nodeIndex with `dataDescription.Parent.IndexOf(dataDescription)`.

Reply:
Done